### ISSUE TITLE
Replication progress sometimes resets on relogging in SSP

### ISSUE THREAD

**Issue by dominx** [simeonradivoev/MatterOverdrive#152]
I'm playing Horizons Daybreaker 1.0.2, Matter Overdrive 0.3.4.1

It happens perhaps once in about ten relogs.
I'm making nether stars for power generation so it's quite painful as it takes almost two hours to replicate one. Really invalidates whole setup.
[My setup](http://i.imgur.com/Tnj0UxP.png) There is constant power supply, replicators are full of matter and there is nothing emiting redstone nearby. I have four of them running and the same happens for all of them.

**Comment by shadowfacts** [comment]
Can you please confirm with the latest version of Matter Overdrive, 0.4.0-RC2?

**Comment by dominx** [comment]
Yes, it still happens. I've tried relogging perhaps 30 times before it happened. I was replicating nether stars. It might have something to do with how much progress has been made so far and the cost of item. IIRC every time it happened it was at least 5 or 6 pixels (the minecraft interface ones not actual) with nether star in all four replicators. I've done testing on a save with replication started in the 0.3.4.1 I couldn't start it in the new because of the other bug I reported. I don't have time to do any more testing today so I cannot confirm it depends on the progress already made. I've tried before with some cheap items and there was no problem.

BTW What happened to matter happily sloshing about in pipes? Was it lag inducing?

**Comment by dominx** [comment]
I've done some more testing with nether stars. The problem persists in 0.4.0-RC2. Sometimes you lose just a pixel, other times it's 1/4, sometimes all.

It's hard to pinpoint conditions when in happens. I've never seen it happen before progress reaches about 1/4 but after that it is still not consistent. What is consistent is that it happens for all replicators started at the same time. 

I've also experienced replicators getting stuck. They seem to draw power but no progress is made.

**Comment by simeonradivoev** [comment]
So you do not loose all the progress just a pixel or 1/4 ? And are you sure the replicators are stuck not just taking a lot of time ?

**Comment by simeonradivoev** [comment]
It also resets the progress of replication if the replicator runs out of power.

**Comment by dominx** [comment]
I've been unable to see any patters in progress loss. It can be all, some, or just a pixel. I'm not even sure how often it's just pixel because it's so easy to miss.

This time I was doing the testing in creative. The replicators were fully powered all the time. I'm not sure about the stuck thing. I was was waiting and relogging a lot. I was observing two replicators. One was gaining progres and sometimes losing on relog, the other showed no progress. But then it started again and I don't know when it happened.
